tests/test_merge.py

```diff
@@ -28,14 +29,34 @@
 def test_merge_opossum_information(opossum_package: OpossumPackage) -> None:
     opossum_information = OpossumInformation(
         Metadata("project-id", "30-05-2023", "test data"),
-        Resource({"A": Resource({"B": Resource({})})}),
+        Resource(
+            ResourceType.TOP_LEVEL,
+            {
+                "A": Resource(
+                    ResourceType.FOLDER, {"B": Resource(ResourceType.FOLDER, {})}
```

Comment:
i would like to see a mix of files and folders here + in the next test, to prove it works, and to make the test more realistic

Reply:
I changed the tests and also fixed the _merge_resource method as it indeed wasn't working as expected for these mixes.